compiler/test/compilable/test21331.sh

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+objfile="${OUTPUT_BASE}${OBJ}"
+$DMD -c -m${MODEL} -of${objfile} ${EXTRA_FILES}/${TEST_NAME}.d
+
+nm ${objfile} | grep -E '16__lambda_L13_C17.+15__lambda_L5_C19'
+nm ${objfile} | grep -E '16__lambda_L13_C17.+17__lambda_L5_C19_1'
+nm ${objfile} | grep -E '18__lambda_L13_C17_1.+15__lambda_L5_C19'
+nm ${objfile} | grep -E '18__lambda_L13_C17_1.+17__lambda_L5_C19_1'
```

Comment:
I gave up on a testcase for the actual issue, wrt. empty tuples in parameter UDAs being 'optimized' later and affecting `toPrettyChars()`.

Instead, I went with this - v2.111 produces these symbols:
```
_D9test21331__T1SSQq3barFZ16__lambda_L13_C17ZQBe3fooMFZ__T15__lambda_L5_C19TiZQuFNaNbNiNfiZi
_D9test21331__T1SSQq3barFZ16__lambda_L13_C17ZQBe3fooMFZ__T17__lambda_L5_C19_1TiZQwFNaNbNiNfiZi
_D9test21331__T1SSQq3barFZ18__lambda_L13_C17_1ZQBg3fooMFZ__T17__lambda_L5_C19_2TiZQwFNaNbNiNfiZi
_D9test21331__T1SSQq3barFZ18__lambda_L13_C17_1ZQBg3fooMFZ__T17__lambda_L5_C19_3TiZQwFNaNbNiNfiZi
```

I.e., the inner lambda got renamed to `_2` and `_3` when the outer one was renamed to `_1`. Using the parent mangle avoids this and makes the inner one start without suffix again.